crates/hud/src/interaction.rs

```diff
@@ -0,0 +1,39 @@
+use bevy::prelude::{Commands, Component, Input, Interaction, MouseButton, Query, ResMut, SystemLabel};
+
+use de_core::objects::BuildingType;
+
+use crate::draft::place_draft;
+use crate::interaction::BuildBuildingButton::BuildBuilding;
+
+#[derive(Component)]
+pub enum BuildBuildingButton {
+    BuildBuilding(BuildingType)
+}
+
+/// Handles what should when then a menu item in the main menu is selected
+/// by the user.
+///
+/// # Arguments
+///
+/// * `query` - The query for the buttons in the menu.
+pub fn handle_item_click(
```

Comment:
I generally try to place all systems alongside the plugin which adds them. Note that `hud` crate implements a PluginGroups which can add multiple plugins. So I suggest to create a new plugin in each module so they are not all added from `plugin.rs`.

Together with that, `plugin.rs` should be renamed to something more specific since it would no longer be the only plugin in the crate.